packages/block-editor/src/store/selectors.js

```diff
@@ -278,6 +278,29 @@ export const getGlobalBlockCount = createSelector(
 	( state ) => [ state.blocks.order, state.blocks.byClientId ]
 );
 
+/**
+ * Returns all global blocks that match a blockName. Results include nested blocks.
+ *
+ * @param {Object}  state     Global application state.
+ * @param {?string} blockName Optional block name, if not specified, returns an empty array.
+ *
+ * @return {Array} Array of clientIds of blocks with name equal to blockName.
+ */
+export const __experimentalGetGlobalBlocksByName = createSelector(
+	( state, blockName ) => {
+		if ( ! blockName ) {
+			return EMPTY_ARRAY;
+		}
+		const clientIds = getClientIdsWithDescendants( state );
+		const foundBlocks = clientIds.filter( ( clientId ) => {
```

Comment:
Could we just return the result of the filter? Do we need to use the `EMPTY_ARRAY` const?

Reply:
Filter will return a new array reference. Thus we use the reference to a constant object reference `EMPTY_ARRAY` so as to avoid returning a new object reference if the selector returns "nothing". It's a minor perf thing but it could make a big difference in certain components.

Reply:
I see Kerry even wrote a test to validate that. See the final `it()` in the tests.